correction-station/correction_station.go

```diff
@@ -286,10 +250,12 @@ func (r *rtkStation) Close(ctx context.Context) error {
 		return err
 	}
 
-	// close the serial port
-	err = r.serialWriter.(io.ReadWriteCloser).Close()
-	if err != nil {
-		return err
+	if r.protocol == serialStr {
+		// close the serial port
+		err = r.serialWriter.(io.ReadWriteCloser).Close()
+		if err != nil {
+			return err
```

Comment:
Let's specify what this error a little bit more. something like "expected protocol to be serial,"

Reply:
This is being split into two separate models in the next PR so going to leave as is for now.